### ISSUE TITLE
Identifiable parentheses  for the parse tree

### ISSUE THREAD

**Issue by walter-weinmann** [K2InformaticsGmbH/sqlparse#68]
The use of parentheses in the following grammar rules should be visible in the parse tree:

```
from_column -> '(' join_clause ')'                                                              : ['$2'].
from_column ->     join_clause                                                                  : ['$1'].

query_partition_clause -> PARTITION BY     scalar_exp_commalist                                 : {partition_by, '$3'} .
query_partition_clause -> PARTITION BY '(' scalar_exp_commalist ')'                             : {partition_by, '$4'}.

query_term -> query_spec                                                                        : '$1'.
query_term -> '(' query_exp ')'                                                                 : '$2'.

search_condition -> '(' search_condition ')'                                                    : '$2'.

```

**Comment by walter-weinmann** [comment]
Suggested solution: The parse tree is extended as follows:

```
from_column -> '(' join_clause ')'                                                              : [{'$2', "("}].
from_column ->     join_clause                                                                  : [{'$1', []}].

query_partition_clause -> PARTITION BY     scalar_exp_commalist                                 : {partition_by, '$3', []} .
query_partition_clause -> PARTITION BY '(' scalar_exp_commalist ')'                             : {partition_by, '$4', "("}.

query_term -> query_spec                                                                        : {'$1', []}.
query_term -> '(' query_exp ')'                                                                 : {'$2', "("}.

search_condition -> '(' search_condition ')'                                                    : {'$2, "("}'.

```

**Comment by walter-weinmann** [comment]
Solved by #91.
